pkg/controller/util.go

```diff
@@ -11,23 +11,19 @@ import (
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 )
 
 const DataVolName = "cdi-data-vol"
 
 // return a pvc pointer based on the passed-in work queue key.
 func (c *Controller) pvcFromKey(key interface{}) (*v1.PersistentVolumeClaim, error) {
-	keyString, ok := key.(string)
-	if !ok {
-		return nil, errors.New("key object not of type string")
-	}
-	obj, ok, err := c.pvcInformer.GetIndexer().GetByKey(keyString)
+	obj, err := objFromKey(c.pvcInformer, key)
```

Comment:
can the type of obj desired be passed to `objFromKey`? That would reduce some of the similar err handling.

Reply:
which error handling do you mean? we are passing in an informer, how would the type of object desired be useful in this case?

Reply:
the informer is a core component of the controller, so `objFromKey` may be more suited as a method

Reply:
ack